mantidproject/mantid
# Add common workbench launcher executable to harmonise startup code.

:warning: Requires testing on all operating systems :warning: 

**Description of work.**

Introduces a common workbench launcher for macOS/Windows written in C++. The original driver was to move away from the executable generated by [PS1-To-Exe](https://gallery.technet.microsoft.com/PS1-To-EXE-Generator-d39e8be2) as

* it can get trapped by some virus scanners, e.g. McAfee 
* it requires the .NET framework and this does not really make sense as a dependency for us

It was discovered that the startup launcher for macOS/Windows were nearly identical so this introduces this executable for macOS too. Harmonizing them should reduce the maintenance burden of having to remember to update multiple locations. Linux has not yet been included as it's launcher is more complicated.

As part of this work the errorreport GUI code has been moved to the `mantidqt` package so that it could be more easily accessible without additional `sys.path` updates within the launcher.

**To test:**

Requires testing on all OSes to confirm nothing is broken.

For both development build and a package:

* Build and test workbench starts as it used to do for that environment. Also start the new `MantidWorkbench` executable in the `bin` directory and check this works.
* Check the error reporter starts on a crash using the `Segfault` algorithm. You'll need to check the `usagereports.enabled=1` setting in the properties file.
* Also check the error reporter pops up for an unhandled exception. 

**Additionally for Visual Studio only.**

The Visual studio build should now run without need for the `visual-studio.bat` file. Run cmake on this branch and then open the solution by opening `Mantid.sln` and **not** `visual-studio.bat`. Build and run workbench.


Fixes #28871 
Fixes #28861
Fixes #29080
Refs #28263

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
